test/Dialect/FIRRTL/lower-types.mlir

```diff
@@ -1034,38 +849,267 @@ module  {
+  }
+}
+// CHECK: firrtl.module @multidimRead(in %a_0_0: !firrtl.uint<2>, in %a_0_1: !firrtl.uint<2>, in %a_0_2: !firrtl.uint<2>, in %a_0_3: !firrtl.uint<2>, in %a_1_0: !firrtl.uint<2>, in %a_1_1: !firrtl.uint<2>, in %a_1_2: !firrtl.uint<2>, in %a_1_3: !firrtl.uint<2>, in %a_2_0: !firrtl.uint<2>, in %a_2_1: !firrtl.uint<2>, in %a_2_2: !firrtl.uint<2>, in %a_2_3: !firrtl.uint<2>, in %a_3_0: !firrtl.uint<2>, in %a_3_1: !firrtl.uint<2>, in %a_3_2: !firrtl.uint<2>, in %a_3_3: !firrtl.uint<2>, in %sel: !firrtl.uint<2>, out %b: !firrtl.uint<2>)
+// CHECK:   %[[SELMUX1:.+]] = firrtl.mux(%[[SELMUX0:.+]], %a_2_0, %a_3_0) : (!firrtl.uint<1>, !firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<2>
+// CHECK:   %[[SELMUX2:.+]] = firrtl.mux(%[[SELMUX0]], %a_2_1, %a_3_1) : (!firrtl.uint<1>, !firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<2>
+// CHECK:   %[[SELMUX3:.+]] = firrtl.mux(%[[SELMUX0]], %a_2_2, %a_3_2) : (!firrtl.uint<1>, !firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<2>
+// CHECK:   %[[SELMUX4:.+]] = firrtl.mux(%[[SELMUX0]], %a_2_3, %a_3_3) : (!firrtl.uint<1>, !firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<2>
+ // CHECK:     %[[SELMUX6:.+]] = firrtl.mux(%[[SELMUX5:.+]], %a_1_0, %[[SELMUX1]]) : (!firrtl.uint<1>, !firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<2>
+ // CHECK:     %[[SELMUX7:.+]] = firrtl.mux(%[[SELMUX5]], %a_1_1, %[[SELMUX2]]) : (!firrtl.uint<1>, !firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<2>
+ // CHECK:     %[[SELMUX8:.+]] = firrtl.mux(%[[SELMUX5]], %a_1_2, %[[SELMUX3]]) : (!firrtl.uint<1>, !firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<2>
+ // CHECK:     %[[SELMUX9:.+]] = firrtl.mux(%[[SELMUX5]], %a_1_3, %[[SELMUX4]]) : (!firrtl.uint<1>, !firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<2>
+ // CHECK:     %[[SELMUX11:.+]] = firrtl.mux(%[[SELMUX10:.+]], %a_0_0, %[[SELMUX6]]) : (!firrtl.uint<1>, !firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<2>
+ // CHECK:     %[[SELMUX12:.+]] = firrtl.mux(%[[SELMUX10]], %a_0_1, %[[SELMUX7]]) : (!firrtl.uint<1>, !firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<2>
+ // CHECK:     %[[SELMUX13:.+]] = firrtl.mux(%[[SELMUX10]], %a_0_2, %[[SELMUX8]]) : (!firrtl.uint<1>, !firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<2>
+ // CHECK:     %[[SELMUX14:.+]] = firrtl.mux(%[[SELMUX10]], %a_0_3, %[[SELMUX9]]) : (!firrtl.uint<1>, !firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<2>
+ // CHECK:     %[[SELMUX16:.+]] = firrtl.mux(%[[SELMUX15:.+]], %[[SELMUX13]], %[[SELMUX14]]) : (!firrtl.uint<1>, !firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<2>
+ // CHECK:     %[[SELMUX18:.+]] = firrtl.mux(%[[SELMUX17:.+]], %[[SELMUX12]], %[[SELMUX16]]) : (!firrtl.uint<1>, !firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<2>
+ // CHECK:     %[[SELMUX20:.+]] = firrtl.mux(%[[SELMUX19:.+]], %[[SELMUX11]], %[[SELMUX18]]) : (!firrtl.uint<1>, !firrtl.uint<2>, !firrtl.uint<2>) -> !firrtl.uint<2>
+ // CHECK:     firrtl.connect %b, %20 : !firrtl.uint<2>, !firrtl.uint<2>
```

Comment:
`CHECK-NEXT` is going to be a tighter check that all of this occurs and nothing in between. Also, consider dropping type information from the test if you feel that's not relevant.

Ditto for other tests below.

Reply:
Yea, lots of test changes since you looked.